src/adaptors/abracadabra/index.js

```diff
@@ -249,9 +250,147 @@ const getApy = async () => {
 
   return pools.flat();
 };
+const ABICDP = {
+  collateral: {
+    inputs: [],
+    name: 'collateral',
+    outputs: [
+      {
+        internalType: 'contract IERC20',
+        name: '',
+        type: 'address',
+      },
+    ],
+    stateMutability: 'view',
+    type: 'function',
+  },
+  COLLATERIZATION_RATE: {
+    inputs: [],
+    name: 'COLLATERIZATION_RATE',
+    outputs: [
+      {
+        internalType: 'uint256',
+        name: '',
+        type: 'uint256',
+      },
+    ],
+    stateMutability: 'view',
+    type: 'function',
+  },
+  totalBorrow: {
+    inputs: [],
+    name: 'totalBorrow',
+    outputs: [
+      {
+        internalType: 'uint128',
+        name: 'elastic',
+        type: 'uint128',
+      },
+      {
+        internalType: 'uint128',
+        name: 'base',
+        type: 'uint128',
+      },
+    ],
+    stateMutability: 'view',
+    type: 'function',
+  },
+};
+
+const cdpChainMap = {
+  1: 'ethereum',
+  43114: 'avalanche',
+  250: 'fantom',
+  42161: 'arbitrum',
+  56: 'binance',
+};
+
+const chainMap = {
+  1: 'ethereum',
+  43114: 'avax',
+  250: 'fantom',
+  42161: 'arbitrum',
+  56: 'bsc',
+};
+const MIM_ID = 'magic-internet-money';
+const getCdp = async () => {
+  const URL = 'https://analytics.abracadabra.money/api/pools';
+  const data = (await utils.getData(URL)).pools;
+  const mimPrice = (await utils.getPrices([`coingecko:${MIM_ID}`]))
+    .pricesByAddress[MIM_ID.toLowerCase()];
+  const result = [];
+  for (const chainId of Object.keys(cdpChainMap)) {
+    const dataByChain = data.filter((e) => e.network == chainId);
+    const chain = cdpChainMap[chainId];
+    console.log(chain);
+    const collateralAddress = (
+      await sdk.api.abi.multiCall({
+        abi: ABICDP.collateral,
+        calls: dataByChain.map((p) => {
+          return { target: p.address };
+        }),
+        chain: chainMap[chainId],
+      })
+    ).output.map((e) => e.output);
+
+    const collateralRate = (
+      await sdk.api.abi.multiCall({
+        abi: ABICDP.COLLATERIZATION_RATE,
+        calls: dataByChain.map((p) => {
+          return { target: p.address };
+        }),
+        chain: chainMap[chainId],
+      })
+    ).output.map((e) => e.output);
+
+    const underlyings = (
+      await sdk.api.abi.multiCall({
+        abi: 'erc20:symbol',
+        calls: collateralAddress.map((target) => {
+          return { target };
+        }),
+        chain: chainMap[chainId],
+        requery: false,
+      })
+    ).output.map((e) => e.output);
+    const coins = collateralAddress.map((address) => `${chain}:${address}`);
+    const prices = (await utils.getPrices(coins)).pricesByAddress;
+
+    const _result = dataByChain.map((pool, index) => {
+      const totalSupplyUsd =
+        Number(pool.totalCollaterel) *
+        prices[collateralAddress[index].toLowerCase()];
```

Comment:
it seems to me the nbs from `'https://analytics.abracadabra.money/api/pools';` are already in usd.